### ISSUE TITLE
Audio recording timing, and Game Reset doesn't always work.

### ISSUE THREAD

**Issue by cdmckinley** [fdekeers/poketools#2]
Hello,

I've cloned the repository locally, and have it mostly working for Arceus resets. The main issue I'm having is that the RESET_GAME macro doesn't always bring up the home menu. Is there a reliable fix for this? 

Something else I had to do was adjust was the timing of the recording, and I don't know how to be sure that it will accurately notice a shiny. How does the audio recognition work? If the shiny sound effect plays during the recording, does the exact timing matter?

I'd also be happy to share my changes for Arceus, though I'd need to do some adjustment to let the program work with Arceus in addition to (as opposed to 'instead of') other scenarios. Let me know.

**Comment by cdmckinley** [comment]
I've also run into this a couple times.

```
Recording game sound...
python3: src/hostapi/alsa/pa_linux_alsa.c:3641: PaAlsaStreamComponent_BeginPolling: Assertion `ret == self->nfds' failed.
Aborted
```

I thought it'd be good to comment as I just made an issue, but can make this a new issue if that's helpful.

**Comment by fdekeers** [comment]
Hi,

For the first issue, the fact that the RESET_GAME macro does not always go back to the Home menu:
The macros timings are quite arbitrary, and as the console can have slightly different behaviours from time to time, it can happen that the timings in the `macros.py` file are too short for the script to correctly reset the game. If you experiment this problem often, please try considering extending the timings that seem to be the problem.

For the timing of the sound recording itself:
I set it arbitrarily to 4 seconds, and it has worked without problems for all the Pokémon I've hunted for now. However, as different Pokémon have different "battle starting scenes", the constants that you might have to adjust are GAME_LOADING_TIME and BATTLE_LOADING_TIME, as well as the input sequence needed to start the battle in the START_BATTLE macro. For now, the tool necessitates those adjustments before starting the hunt for a different Pokémon.

For your issue related to alsa:
I remember running into that a couple of times, but don't remember what caused it...
Make sure all the correct libraries are installed (libasound-dev, portaudio19-dev, libatlas-base-dev, and I think another one is necessary on RasPi) and that the console is correctly connected with an audio cable to your machine and considered as a audio source.
Which machine are you running the script on ?
 
Thanks for using our tool !

**Comment by fdekeers** [comment]
Hi again,

I extended your idea of having different scenarios.
Now, the scenarios timings and macros are defined in individual files, and one simply has to give the name of the file as command line argument to use it.
I thus moved the Arceus-related config to the `arceus.py` file.

Thanks again for your contribution !

**Comment by cdmckinley** [comment]
This looks really good!

I've been running this script with Arceus a lot, and I realized the timing isn't very consistent. So I've been making timing adjustments again. I plan to make a PR soon, but the exact timing will depend on whether we use 'X' to close the game, and then start it again, or 'Y' to restart the game, which seems to be faster. Maybe you know of an issue with 'Y' that I don't, though.

Let me know which button you'd prefer to use, and I'll implement it using that button.

Glad to help, and thanks for making the initial project!

**Comment by fdekeers** [comment]
I am surprised, I did not know that the Y button can be used to restart the game.
I tried on my console and it did not seem to work.
Can you tell me exactly how it should be used ?

**Comment by cdmckinley** [comment]
Upon some further research, it looks like this only happens when user-selection before a game is currently enabled. This can be disabled if there's only one user, which I would assume also saves time for anyone with this setting.

Maybe accounting for a single user, or multiple, on a system could be another option?

**Comment by fdekeers** [comment]
I created a new user to test that, and indeed the Y button is used to switch between users. However, it does not seem faster than the normal reset with the X button.
Anyway, I will add the necessary code in the macros for multiple-users consoles, and a command line argument to provide the index number of the user that must be selected.
